redisvl/extensions/session_manager/semantic_session.py

```diff
@@ -0,0 +1,364 @@
+from time import time
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+from redis import Redis
+
+from redisvl.extensions.session_manager import BaseSessionManager
+from redisvl.index import SearchIndex
+from redisvl.query import FilterQuery, RangeQuery
+from redisvl.query.filter import Num, Tag
+from redisvl.redis.utils import array_to_buffer
+from redisvl.schema.schema import IndexSchema
+from redisvl.utils.vectorize import BaseVectorizer, HFTextVectorizer
+
+
+class SemanticSessionManager(BaseSessionManager):
+    session_field_name: str = "session_tag"
+    user_field_name: str = "user_tag"
+    vector_field_name: str = "vector_field"
+
+    def __init__(
+        self,
+        name: str,
+        session_tag: str,
+        user_tag: str,
+        prefix: Optional[str] = None,
+        vectorizer: Optional[BaseVectorizer] = None,
+        distance_threshold: float = 0.3,
+        redis_client: Optional[Redis] = None,
+    ):
+        """Initialize session memory with index
+
+        Session Manager stores the current and previous user text prompts and
+        LLM responses to allow for enriching future prompts with session
+        context. Session history is stored in prompt:response pairs referred to
+        as exchanges.
+
+        Args:
+            name (str): The name of the session manager index.
+            session_tag (str): Tag to be added to entries to link to a specific
+                session.
+            user_tag (str): Tag to be added to entries to link to a specific user.
+            prefix (Optional[str]): Prefix for the keys for this session data.
+                Defaults to None and will be replaced with the index name.
+            vectorizer (Vectorizer): The vectorizer to create embeddings with.
+            distance_threshold (float): The maximum semantic distance to be
+                included in the context. Defaults to 0.3.
+            redis_client (Optional[Redis]): A Redis client instance. Defaults to
+                None.
+
+        The proposed schema will support a single vector embedding constructed
+        from either the prompt or response in a single string.
+
+        """
+        super().__init__(name, session_tag, user_tag)
+
+        prefix = prefix or name
+
+        if vectorizer is None:
+            self._vectorizer = HFTextVectorizer(
```

Comment:
It would seem that there's no way to actually set your own vectorizer here. What if a user providers a `vectorizer` object?

Reply:
That's a mistake. Nice catch.